### PR TITLE
Cachegroup fallbacks deprecation

### PR DESCRIPTION
**Pull request by ocket8888**
## What does this PR (Pull Request) do?

- [x] This PR is not related to any Issue

Adds a deprecation notice to the `cachegroup_fallbacks` TO API endpoint documentation, as discussed in the dev mailing list. Also added missing documentation for the `fallbacks` property of API request/response payloads to/from the `cachegroups` and `cachegroups/{{ID}}` endpoints. Plus I did a little cleaning up of unnamed footnotes and unnecessary escape sequences.

## Which Traffic Control components are affected by this PR?

- Documentation

## What is the best way to verify this PR?
Build and read the documentation.

## The following criteria are ALL met by this PR
- [x] Tests are unnecessary
- [x] This PR includes documentation
- [x] This PR includes an update to CHANGELOG.md
- [x] This PR includes any and all required license headers
- [x] This PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY**

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/4075/
Test PASSed.

File: docs/source/api/cachegroup_fallbacks.rst [hunk 1]
```diff
@@ -19,6 +19,10 @@
 ``cachegroup_fallbacks``
 ************************
 
+.. deprecated:: ATCv4.0
```
Comment (mhoppa) [inline docs/source/api/cachegroup_fallbacks.rst]: I have seen in other deprecation notices that we use the API version not the ATC version. Should this match those?

File: docs/source/api/cachegroups.rst [hunk 2]
```diff
@@ -123,19 +130,25 @@ Creates a :term:`Cache Group`
 
 Request Structure
 -----------------
-:fallbackToClosest: If ``true``, the Traffic Router will fall back on the 'closest' :term:`Cache Group` to this one, when this one fails
+:fallbacks: An optional field which, when present, should contain an array of names of other :term:`Cache Groups` on which the Traffic Router will fall back in the event that this :term:`Cache Group` fails/becomes unavailable\ [#fallbacks]_
+
+	.. versionadded:: ATCv4.0
```
Comment (mhoppa) [inline docs/source/api/cachegroups.rst]: should this be API version? same comment in the other places that reference this version

File: docs/source/api/cachegroup_fallbacks.rst (hunk 1 above)
Comment (ocket8888) [reply docs/source/api/cachegroup_fallbacks.rst]: See below.

File: docs/source/api/cachegroups.rst (hunk 2 above)
Comment (ocket8888) [reply docs/source/api/cachegroups.rst]: Unfortunately no. The `fallbacks` array was added to the cachegroup response object in all API versions in ATCv4.

FWIW this is a big part of why I don't think the API ought to be versioned separately from TO.

Comment (mhoppa) [reply docs/source/api/cachegroups.rst]: 💯 agree very confusing to end users

**Comment by ocket8888** [comment]
> _The cachegroup_fallbacks endpoints should have a deprecation message added to its response_

done.

**Comment by ocket8888** [comment]
retest this please

**Comment by ocket8888** [comment]
retest this please

**Comment by ocket8888** [comment]
retest this please

**Comment by ocket8888** [comment]
retest this please

**Comment by ocket8888** [comment]
retest this please

**Pull request merged**